### DISCUSSION TITLE
vscode xmake调试的时候无法进行标准输入输出，并且默认调试参数无法正常使用

### DISCUSSION THREAD

**Discussion by wanzzhehe** [xmake-io/xmake-vscode#187] [Q&A] [answered]
<img width="1286" alt="image" src="https://user-images.githubusercontent.com/26498686/232265518-142ba930-fbcf-4941-9b01-71bedc526591.png">

我在下面和设置里面没有找到相关选项

**Comment by wanzzhehe** [comment]
嘶……现在又可以有输出了，所以感觉可能是vscode出了什么问题

**Comment by wanzzhehe** [comment]
<img width="1281" alt="image" src="https://user-images.githubusercontent.com/26498686/232314199-9aad80a7-3a68-4e29-b63c-f1b417c20e25.png">
好，现在又出现新的问题了

**Reply by waruqi** [reply]
看着好像跟 xmake 插件没啥关系，这个错误似乎是 launch.json 里面的 json 格式无效导致无法解析。。但是 xmake 插件基本没去用 launch.json ，直接内部传 lauch object 给的调试器，调试时候不会用外面的 launch.json。。

你可以自己编辑 launch.json 看看有什么错误。是不是自己外面 launch.json 里还配置了啥？

**Reply by wanzzhehe** [reply]
<img width="1286" alt="image" src="https://user-images.githubusercontent.com/26498686/232353904-04f83b13-405b-4b10-974e-2362046d0c8a.png">
问题是我都没有用launch.json啊

而且只要我把 "default" 字段给注释掉后就能正常启动调试。

**Reply by waruqi** [reply, marked as answer]
因为你配置的格式不对，解析失败。。default 指的是默认的运行 target 的参数列表。你里面又去套了个 client target 。。json array 变成了 json dictionary

```
{
    "xmake.debuggingTargetsArguments": {
        "default": [],
        "client": ["127.0.0.1"]
    }
}
```

如果只是默认目标，只需要


```
{
    "xmake.debuggingTargetsArguments": {
        "default": ["127.0.0.1"]
    }
}
```

配置项那有提示的。。仔细看下
<img width="526" alt="image" src="https://user-images.githubusercontent.com/151335/232358076-80e1b466-7c55-4641-85b7-0240532c9d45.png">

**Reply by wanzzhehe** [reply]
<img width="676" alt="image" src="https://user-images.githubusercontent.com/26498686/233758840-37ea1558-b334-4b01-aaa3-43d34402b678.png">
当时你写的Orz

**Reply by waruqi** [reply]
> <img alt="image" width="676" src="https://user-images.githubusercontent.com/26498686/233758840-37ea1558-b334-4b01-aaa3-43d34402b678.png">
> 
> 当时你写的Orz

这个贴的是插件源码里面的参数描述配置，它这个格式是用来描述插件本身的，不是用户端的格式。。只是为了告诉你 有这个配置项可以用。。

而且具体传参格式，后面也有举例，你仔细看。。

<img width="270" alt="image" src="https://user-images.githubusercontent.com/151335/233759452-995805a4-cf67-404a-b0d4-98148fb8c31c.png">

这里也有

<img width="264" alt="image" src="https://user-images.githubusercontent.com/151335/233759574-9f430557-dacb-4c55-ab39-a419554c2a76.png">

**Reply by wanzzhehe** [reply]
> > <img alt="image" width="676" src="https://user-images.githubusercontent.com/26498686/233758840-37ea1558-b334-4b01-aaa3-43d34402b678.png">
> > 当时你写的Orz
> 
> 这个贴的是插件源码里面的参数描述配置，它这个格式是用来描述插件本身的，不是用户端的格式。。只是为了告诉你 有这个配置项可以用。。
> 
> 而且具体传参格式，后面也有举例，你仔细看。。
> 
> <img alt="image" width="270" src="https://user-images.githubusercontent.com/151335/233759452-995805a4-cf67-404a-b0d4-98148fb8c31c.png">
> 
> 这里也有
> 
> <img alt="image" width="264" src="https://user-images.githubusercontent.com/151335/233759574-9f430557-dacb-4c55-ab39-a419554c2a76.png">

我只能说是我误会了Orz，我以为是 "default" 里面的格式
